### ISSUE TITLE
New features

### ISSUE THREAD

**Issue by drkoaeg** [Lunatrius/Config-Patcher#1]
Would you make the patching undoable?
And another point: Could you not only trigger patching at start-up, but also implement option of time (reading out game stats json game time)?

Thanks for thinking about this.

**Comment by Lunatrius** [comment]
The patching itself is already done in a rather unsafe way and any mod not reading and writing the config in a standard way might break. It would require keeping a history of all previously used values in order to be able to undo a patch operation.

I'm not entirely sure that I understand what you mean with the time option. Patching can only be done on start up, before the standard config related event is triggered. Triggering it at a later stage of the loading process and/or while in-game might not properly apply the patches.

**Comment by drkoaeg** [comment]
Thank you for this explanation. I was making these proposals by trying to use this mod for another effect:
I use mods that change seasons, so I was hoping to adapt other mods to this season system with your modification. So to make it very clear, for example I wanted to change leave falling config of better foliage mod to only be active in autumn. 
Because of these slow time changes, I thought it would not be a problem, if it changes not until next start-up. But I would need a trigger for this change.

**Comment by Lunatrius** [comment]
You wouldn't be able to "tell" Config Patcher to undo a patch though... you'd have to replace the patches as it is right now.

The feature that you're requesting might be better off as a minetweak-like mod.

**Comment by drkoaeg** [comment]
Okay, I see, thank you.
